sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -493,6 +493,11 @@ case class Cast(
 
   final override def nodePatternsInternal(): Seq[TreePattern] = Seq(CAST)
 
+  override def contextIndependentFoldable: Boolean = {
+    child.contextIndependentFoldable && !DataTypeUtils.containsTimestampOrUDT(dataType) &&
```

Comment:
I don't think UDT matters here. The CAST implementation only allows pass-through of UDT casting, there is no operation and no context is needed.

Reply:
ok, updated